providers/google/src/airflow/providers/google/cloud/hooks/gen_ai.py

```diff
@@ -194,3 +197,182 @@ def create_cached_content(
+        """
+        client = self.get_genai_client()
+        resp = client.batches.create(
+            model=model,
+            src=source,
+            config=create_batch_job_config,
+        )
+        return resp
+
+    def delete_batch_job(
+        self,
+        job_name: str,
+    ):
+        """
+        Delete batch job using Gemini Batch API.
+
+        :param job_name: Required. Batch job name.
+        """
+        client = self.get_genai_client()
+        client.batches.delete(name=job_name)
+
+    def cancel_batch_job(
+        self,
+        job_name: str,
+    ):
+        """
+        Cancel batch job using Gemini Batch API.
+
+        :param job_name: Required. Batch job name.
+        """
+        client = self.get_genai_client()
+        client.batches.cancel(
+            name=job_name,
+        )
+
+    def create_embeddings(
+        self,
+        model: str,
+        source: dict | str,
+        create_embeddings_config: CreateBatchJobConfig | dict | None = None,
+    ) -> BatchJob:
+        """
+        Create batch job for embeddings using Gemini Batch API to process large-scale, non-urgent tasks.
+
+        :param model: Required. Gemini model name to process requests.
+        :param source: Required. Requests that will be sent to chosen model.
+            Can be in format of Inline requests or file name.
+        :param create_embeddings_config: Optional. Configuration parameters for embeddings batch job.
+        """
+        client = self.get_genai_client()
+        input_type = "inlined_requests"
+
+        if isinstance(source, str):
+            input_type = "file_name"
+
+        self.log.info("Using %s to create embeddings", input_type)
+
+        resp = client.batches.create_embeddings(
+            model=model,
+            src={input_type: source},
+            config=create_embeddings_config,
+        )
+        return resp
+
+    def upload_file(self, path_to_file: str, upload_file_config: dict | None = None):
+        """
+        Upload file for batch job or embeddings batch job using Gemini Files API.
+
+        :param path_to_file: Required. Path to file on local filesystem.
+        :param upload_file_config: Optional. Configuration for file upload.
+        """
+        client = self.get_genai_client()
+
+        if upload_file_config is None:
+            self.log.info("Default configuration will be used to upload file")
+            try:
+                file_name, file_type = path_to_file.split("/")[-1].split(".")
+                upload_file_config = {"display_name": file_name, "mime_type": file_type}
+            except ValueError as exc:
+                raise ValueError(
+                    "Error during unpacking file name or mime type. Please check file path"
+                ) from exc
+
+        resp = client.files.upload(
+            file=path_to_file,
+            config=upload_file_config,
+        )
+        return resp
+
+    def get_file(self, file_name: str):
```

Comment:
Could we add return's value type as we have in Client https://github.com/googleapis/python-genai/blob/main/google/genai/files.py#L271 ?